arangod/Aql/AqlItemBlock.cpp

```diff
@@ -724,30 +725,41 @@ void AqlItemBlock::toVelocyPack(size_t from, size_t to,
   result.add("raw", raw.slice());
 }
 
-void AqlItemBlock::rowToSimpleVPack(size_t const row, velocypack::Options const* options,
+void AqlItemBlock::rowToSimpleVPack(size_t row, velocypack::Options const* options,
                                     arangodb::velocypack::Builder& builder) const {
-  VPackArrayBuilder rowBuilder{&builder};
+  {
+    VPackArrayBuilder rowBuilder{&builder};
 
-  if (isShadowRow(row)) {
-    getShadowRowDepth(row).toVelocyPack(options, *rowBuilder, false);
-  } else {
-    AqlValue{AqlValueHintNull{}}.toVelocyPack(options, *rowBuilder, false);
-  }
-  for (RegisterId reg = 0; reg < getNrRegs(); ++reg) {
-    getValueReference(row, reg).toVelocyPack(options, *rowBuilder, false);
+    if (isShadowRow(row)) {
+      getShadowRowDepth(row).toVelocyPack(options, builder, false);
+    } else {
+      builder.add(VPackSlice::nullSlice());
+    }
+    RegisterId const n = getNrRegs();
+    for (RegisterId reg = 0; reg < n; ++reg) {
+      AqlValue const& ref = getValueReference(row, reg);
+      if (ref.isEmpty()) {
+        builder.add(VPackSlice::noneSlice());
+      } else {
+        ref.toVelocyPack(options, builder, false);
+      }
+    }
   }
 }
 
 void AqlItemBlock::toSimpleVPack(velocypack::Options const* options,
                                  arangodb::velocypack::Builder& builder) const {
-  VPackObjectBuilder block{&builder};
-  block->add("nrItems", VPackValue(size()));
-  block->add("nrRegs", VPackValue(getNrRegs()));
-  block->add(VPackValue("matrix"));
   {
```

Comment:
This is not a cleanup problem. We have seen issues in the past with compilers applying return value optimization and such, so that the closing of the array/object was performed at different times than expected.
However, this function does not return anything, so it is indeed unnecessary to have the extra scope. Will remove it.